ui/decredmaterial/container.go

```diff
@@ -0,0 +1,81 @@
+package decredmaterial
+
+import (
+	"gioui.org/layout"
+	"gioui.org/unit"
+)
+
+type Container struct {
+	container *layout.List
+	scrollbar *Scrollbar
+
+	contentHeight float32
+}
+
+func (t *Theme) Container() *Container {
+	return &Container{
+		container: &layout.List{
+			Axis: layout.Vertical,
+		},
+		scrollbar: t.Scrollbar(),
+	}
+}
+
+func (c *Container) calculateContentHeight(gtx layout.Context, w []func(gtx C) D) {
+	height := float32(0)
+	for i := range w {
+		index := i
+		(&layout.List{Axis: layout.Vertical}).Layout(gtx, 1, func(gtx C, i int) D {
+			dim := layout.UniformInset(unit.Dp(5)).Layout(gtx, w[index])
+			height += float32(dim.Size.Y)
+			return layout.Dimensions{}
+		})
+	}
+	c.contentHeight = height
+}
+
+func (c *Container) Layout(gtx layout.Context, w []func(gtx C) D) layout.Dimensions {
+	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			c.calculateContentHeight(gtx, w)
+			return layout.Dimensions{}
+		}),
+		layout.Rigid(func(gtx C) D {
+			return c.layout(gtx, w)
+		}),
+	)
+}
+
+func (c *Container) layout(gtx layout.Context, w []func(gtx C) D) layout.Dimensions {
+	var visibleFraction, scrollDepth float32
+
+	flexChildren := []layout.FlexChild{
+		layout.Flexed(0.98, func(gtx C) D {
+			if scrolled, progress := c.scrollbar.Scrolled(); scrolled {
+				c.container.Position.First = int(float32(len(w)) * progress)
+			}
+
+			var totalVisibleHeight float32
+
+			dims := c.container.Layout(gtx, len(w), func(gtx C, i int) D {
+				dim := layout.UniformInset(unit.Dp(5)).Layout(gtx, w[i])
+				totalVisibleHeight += float32(dim.Size.Y)
+				return dim
+			})
+
+			visibleFraction = totalVisibleHeight / c.contentHeight
+			scrollDepth = float32(c.container.Position.First) / float32(len(w))
+
+			return dims
+		}),
+		layout.Flexed(0.02, func(gtx C) D {
+			return layout.Inset{
+				Left: unit.Dp(4),
+			}.Layout(gtx, func(gtx C) D {
+				return c.scrollbar.Layout(gtx, scrollDepth, visibleFraction)
+			})
+		}),
+	}
+
+	return layout.Flex{Axis: layout.Horizontal, Spacing: layout.SpaceBetween}.Layout(gtx, flexChildren...)
```

Comment:
This would be better with a stack layout. Using a flex layout would create dimension issues on different screen sizes. 

In the screenshot below, the scrollbar has a width that changes depending on the window size. If I make the window small enough, the scrollbar disappears. Also The scroll bar doesn't seem to touch the bottom of the window when I reach the end of the content.
Screenshot of scrollbar on testing page:
<img width="1202" alt="Screenshot 2020-09-23 at 19 26 19" src="https://user-images.githubusercontent.com/15804688/94053809-b6335b80-fdd2-11ea-80b9-a337c9f341e3.png">